PyPy not supported

## Description

1. Attempting to install `Mathics-omnibus` gets stuck at `Installing build dependencies ... |`.
2. Attempting to install `Mathics3` raises the following error:
```
gcc -pthread -DNDEBUG -O2 -fPIC -I/home/hyperswine/.pyenv/versions/pypy3.9-7.3.10/include/pypy3.9 -c lib/recordclass/_litelist.c -o build/temp.linux-x86_64-3.9/lib/recordclass/_litelist.o
lib/recordclass/_litelist.c: In function ‘litelist_resize’:
lib/recordclass/_litelist.c:31:56: error: ‘PyLiteListObject’ {aka ‘PyListObject’} has no member named ‘ob_item
```

## How to Reproduce

1. From a popos (ubuntu 22.04) environment with pypy 3.9.15, run `pip install Mathics-omnibus`. And get the error in `1`
2. From the same environment, run `pip install Mathics3`. And get the error in `2`

## Expected behavior

Not get stuck at installing build dependencies and not raise an error. I did check the `recordclass` source code, and it seems like the `PyLiteListObject` struct did not in fact contain `ob_item`. Unless Im seeing things, that does seem like a problem with that dependency, so I actually don't know why `Mathics3` works when its including that dep.

## Your Environment

- PopOS 22.04, Intel + Nvidia
- Pypy 3.9.15 through pyenv
- No other mathematica /mathics installs on the system

It's possible to install the latest master of mathics in pypy now that llvmlite is an optional dependency. It mostly works fine as far as I can tell, though a few tests are failing, and I see some warnings about some builtins failing to load when running the CLI:

```
FAILED test/builtin/numbers/test_calculus.py::test_findminimum[MatchQ[FindMinimum[Cos[x]^2, {x,1.}, Method->brent],{_Real,{x->_Real}}]-True--None] - AssertionError
FAILED test/builtin/numbers/test_calculus.py::test_findminimum[MatchQ[FindMaximum[Cos[x]^2, {x,1.}, Method->brent ], {_Real,{x->_Real}}]-True--None] - AssertionError
FAILED test/builtin/numbers/test_calculus.py::test_findminimum[MatchQ[FindMinimum[Cos[x]^2, {x,1.}, Method->golden],{_Real,{x->_Real}}]-True--None] - AssertionError
FAILED test/builtin/numbers/test_calculus.py::test_findminimum[MatchQ[FindMaximum[Cos[x]^2, {x,1.}, Method->golden ], {_Real,{x->_Real}}]-True--None] - AssertionError
FAILED test/builtin/numbers/test_calculus.py::test_findroot[MatchQ[FindRoot[Cos[x+1.5], {x,.1}, Method->brenth], {x->_Real}]-True-] - AssertionError
FAILED test/builtin/numbers/test_nintegrate.py::test_nintegrate[NIntegrate[1., {x,0,1}, Method->"Quadrature"]-1.-Check that the library is already loaded.-messages0] - AssertionError: expected 0; got 1. Messages: ['The Method option should be a built-in method name in {`Automatic`, `Internal`, `Simpson`}. Using...
FAILED test/builtin/numbers/test_nintegrate.py::test_nintegrate[NIntegrate[1., {x,0,1}, Method->"NotAMethod"]-1.-None-messages1] - AssertionError:  do not match.
FAILED test/test_main.py::test_cli - AssertionError: assert None
```

```
    Not able to load mathics.builtin.scipy_utils.integrators. Check your installation.
    mathics.builtin loads from /home/david/git/mathics-core/mathics/core/load

    Not able to load mathics.builtin.scipy_utils.optimizers. Check your installation.
    mathics.builtin loads from /home/david/git/mathics-core/mathics/core/load
```

--- a/mathics/builtin/drawing/plot.py
+++ b/mathics/builtin/drawing/plot.py
@@ -316,7 +316,7 @@ class _ListPlot(Builtin, ABC):
         all_points = (
             points.value
             if hasattr(points, "value") and points.value is not None
-            else eval_N(points, evaluation).to_python()
+            else eval_N(points, evaluation).to_python()  # TODO: force tuple-ness?
         )
 
         # FIXME: arrange for self to have a .symbolname property or attribute

--- a/mathics/builtin/scipy_utils/integrators.py
+++ b/mathics/builtin/scipy_utils/integrators.py
@@ -1,9 +1,8 @@
 # -*- coding: utf-8 -*-
 
 from mathics.core.builtin import check_requires_list
-from mathics.core.util import IS_PYPY
 
-if IS_PYPY or not check_requires_list(["scipy", "numpy"]):
+if not check_requires_list(["scipy", "numpy"]):
     raise ImportError
 
 import numpy as np

--- a/mathics/builtin/scipy_utils/optimizers.py
+++ b/mathics/builtin/scipy_utils/optimizers.py
@@ -5,10 +5,9 @@ from mathics.core.convert.function import expression_to_callable_and_args
 from mathics.core.evaluation import Evaluation
 from mathics.core.expression import Expression
 from mathics.core.systemsymbols import SymbolAutomatic, SymbolFailed, SymbolInfinity
-from mathics.core.util import IS_PYPY
 from mathics.eval.nevaluator import eval_N
 
-if IS_PYPY or not check_requires_list(["scipy", "numpy"]):
+if not check_requires_list(["scipy", "numpy"]):
     raise ImportError
 
 

--- a/mathics/eval/drawing/plot.py
+++ b/mathics/eval/drawing/plot.py
@@ -134,6 +134,7 @@ def compile_quiet_function(expr, arg_names, evaluation, list_is_expected: bool):
 
 
 def eval_ListPlot(
+    # TODO: plot_groups should be a tuple only?
     plot_groups: Union[list, tuple],
     x_range: list,
     y_range: list,

--- a/mathics/eval/drawing/plot.py
+++ b/mathics/eval/drawing/plot.py
@@ -134,7 +134,7 @@ def compile_quiet_function(expr, arg_names, evaluation, list_is_expected: bool):
 
 
 def eval_ListPlot(
-    plot_groups: list,
+    plot_groups: Union[list, tuple],
     x_range: list,
     y_range: list,
     is_discrete_plot: bool,

--- a/mathics/eval/nevaluator.py
+++ b/mathics/eval/nevaluator.py
@@ -29,7 +29,7 @@ def eval_N(
     expression: BaseElement,
     evaluation: Evaluation,
     prec: BaseElement = SymbolMachinePrecision,
-) -> BaseElement:
+) -> Optional[BaseElement]:
     """
     Equivalent to Expression(SymbolN, expression).evaluate(evaluation)
     """

